### PR TITLE
Use a mockable authenticator instead of service swapping

### PR DESCRIPTION
**Pull request by derrabus**
Swapping services is really not fun with the Symfony container. With this PR, I tried to get around swapping services for authentication. WDYT?

Related to #618.

### CODE REVIEW

**Comment by chartjes** [comment]
I *really* like this

**Comment by mdwheele** [comment]
:heart: :heart_eyes: :heart:

File: classes/Infrastructure/Auth/MockableAuthenticator.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Infrastructure\Auth;
+
+use OpenCFP\Domain\Services\Authentication;
+use OpenCFP\Domain\Services\NotAuthenticatedException;
+
+class MockableAuthenticator implements Authentication
```
Comment (BackEndTea) [inline classes/Infrastructure/Auth/MockableAuthenticator.php]: I really like this, but maybe it should be moved to the ```OpenCfp\Test\Helper``` namespace?

Comment (chartjes) [reply classes/Infrastructure/Auth/MockableAuthenticator.php]: 👍 to that suggestion

Comment (derrabus) [reply classes/Infrastructure/Auth/MockableAuthenticator.php]: All right, I will move it.

**Pull request merged**